### PR TITLE
Adding a flag to linker gen script to move rodata to dmem;

### PR DESCRIPTION
**Pull request by tommydcjung**
- Adding a flag --move_rodata_to_dmem.
- By default, this flag is turned off.

### CODE REVIEW

File: software/py/bsg_manycore_link_gen.py [hunk 1]
```diff
@@ -117,6 +118,54 @@ def _section(self, name, address, vma_region, lma_region,
...
+      for i, m in enumerate(section_map):
+        sec = m[0]
+        if sec == ".dmem":
+          m[1].append(".rodata")
+          m[1].append(".rodata*")
+          m[1].append(".srodata.cst16")
+          m[1].append(".srodata.cst8")
+          m[1].append(".srodata.cst4")
+          m[1].append(".srodata.cst2")
+          m[1].append(".srodata*")
```
Comment (gaozihou) [inline software/py/bsg_manycore_link_gen.py:164]: I can confirm the script is working perfectly in both cases. Maybe simplify the codes with the following?

```
      popped_rodata_dram_map = section_map.pop(ro_idx)

      # add to .dmem
      for i, m in enumerate(section_map):
        sec = m[0]
        if sec == ".dmem":
          for in_sec in popped_rodata_dram_map[1]:
            m[1].append(in_sec)
```

File: software/py/bsg_manycore_link_gen.py [hunk 2]
```diff
@@ -44,11 +44,12 @@ class bsg_manycore_link_gen:
...
     + " BSG Manycore Linker Script \n\n"
 
-  def __init__(self, default_data_loc, dram_size, imem_size, sp):
+  def __init__(self, default_data_loc, dram_size, imem_size, sp, move_rodata_to_dmem):
     self._default_data_loc = default_data_loc
     self._dram_size = dram_size
     self._imem_size = imem_size
     self._sp = sp
+    self._move_rodata_to_dmem = move_rodata_to_dmem
     self._opening_comment += \
```
Comment (gaozihou) [inline software/py/bsg_manycore_link_gen.py:53]: Maybe add a print comment for the new parameter?

File: software/py/bsg_manycore_link_gen.py (hunk 1 above)
Comment (tommydcjung) [reply software/py/bsg_manycore_link_gen.py]: okay addressed!

File: software/py/bsg_manycore_link_gen.py (hunk 2 above)
Comment (tommydcjung) [reply software/py/bsg_manycore_link_gen.py:53]: okay addressed!

**Review by gaozihou — APPROVED** [review]
LGTM! I can confirm

- When `LINK_GEN_MOVE_RODATA_TO_DMEM=0` or not set, generated linker is identical to before (except for the new comment)
- When `LINK_GEN_MOVE_RODATA_TO_DMEM=1`, rodata is moved to dmem
- In both cases RTL simulation is running correctly

**Pull request merged**